cranelift/codegen/src/machinst/isle.rs

```diff
@@ -780,6 +780,24 @@ macro_rules! isle_prelude_methods {
             regs.regs()[idx]
         }
 
+        fn abi_arg_slot_regs(&mut self, arg: &ABIArg) -> Option<WritableValueRegs> {
+            match arg {
+                &ABIArg::Slots { ref slots, .. } => match slots.len() {
+                    1 => {
+                        let a = self.temp_writable_reg(slots[0].get_type());
+                        Some(WritableValueRegs::one(a))
+                    }
+                    2 => {
+                        let a = self.temp_writable_reg(slots[0].get_type());
+                        let b = self.temp_writable_reg(slots[1].get_type());
+                        Some(WritableValueRegs::two(a, b))
+                    }
+                    _ => None,
```

Comment:
Let's have a `panic` here instead: in theory we could have a larger number if we have a narrower machine (`I128` on a 32-bit architecture) or larger type. Otherwise we just fail isel in a hard-to-debug way...